pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/service/admin/UserAdminTestUtil.java

```diff
@@ -0,0 +1,404 @@
+            try (PreparedStatement stmt = conn.prepareStatement("SELECT ar.role_id, ar.name FROM access_role ar")) {
+                try (ResultSet rs = stmt.executeQuery()) {
+                    while (rs.next()) {
+                        roles.put(rs.getString(2), rs.getInt(1));
+                    }
+                }
+            } catch (SQLException e) {
+                throw new DsmInternalError("Error getting roles", e);
+            }
+            return roles;
+        });
+    }
+
+    private static int createTestInstance(String instanceName, int studyGroupId) {
+        int instanceId = createInstance(instanceName);
+        SimpleResult res = inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_INSERT_DDP_INSTANCE_GROUP, Statement.RETURN_GENERATED_KEYS)) {
+                stmt.setInt(1, instanceId);
+                stmt.setInt(2, studyGroupId);
+                int result = stmt.executeUpdate();
+                if (result != 1) {
+                    dbVals.resultException = new DsmInternalError("Result count was " + result);
+                }
+                try (ResultSet rs = stmt.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getInt(1);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
+        if (res.resultException != null) {
+            _deleteInstance(instanceId);
+            throw new DsmInternalError("Error adding DDP instance group " + instanceName, res.resultException);
+        }
+        return instanceId;
+    }
+
+    private static int createInstance(String instanceName) {
+        SimpleResult res = inTransaction(conn -> {
+            SimpleResult dbVals = new SimpleResult();
+            try (PreparedStatement stmt = conn.prepareStatement(SQL_INSERT_DDP_INSTANCE, Statement.RETURN_GENERATED_KEYS)) {
+                stmt.setString(1, instanceName);
+                int result = stmt.executeUpdate();
+                if (result != 1) {
+                    dbVals.resultException = new DsmInternalError("Result count was " + result);
+                }
+                try (ResultSet rs = stmt.getGeneratedKeys()) {
+                    if (rs.next()) {
+                        dbVals.resultValue = rs.getInt(1);
+                    }
+                }
+            } catch (SQLException ex) {
+                dbVals.resultException = ex;
+            }
+            return dbVals;
+        });
+
+        if (res.resultException != null) {
+            throw new DsmInternalError("Error adding DDP instance group " + instanceName, res.resultException);
+        }
+        return (int) res.resultValue;
+    }
+
+    private static void deleteInstance(int instanceId) {
```

Comment:
I think this should be `deleteInstanceGroup`?  I was trying to figure out the difference between this method and the one on L392.

Reply:
It actually does both instance and group. I'll name appropriately.